icepyx/core/read.py

```diff
@@ -367,20 +386,53 @@ def __init__(
             assert pattern_ck
             self._filelist = filelist
         elif isinstance(data_source, list):
+            # if data_source is a list pass that directly to _filelist
             self._filelist = data_source
         elif os.path.isdir(data_source):
+            # if data_source is a directory glob search the directory and assign to _filelist
             data_source = os.path.join(data_source, "*")
             self._filelist = glob.glob(data_source, **glob_kwargs)
+        elif isinstance(data_source, str):
+            if data_source.startswith('s3'):
+                # if the string is an s3 path put it in the _filelist without globbing
+                self._filelist = [data_source]
+            else:
+                # data_source is a globable string
+                self._filelist = glob.glob(data_source, **glob_kwargs)
         else:
-            self._filelist = glob.glob(data_source, **glob_kwargs)
-        # Remove any directories from the list
+            raise TypeError(
+                'data_source should be a list of files, a directory, the path to a file, '
+                'or a glob string.'
+            )
+        # Remove any directories from the list (these get generated during recursive
+        # glob search)
         self._filelist = [f for f in self._filelist if not os.path.isdir(f)]
 
         # Create a dictionary of the products as read from the metadata
         product_dict = {}
-        for file_ in self._filelist:
-            product_dict[file_] = is2ref.extract_product(file_)
-
+        self.is_s3 = [False]*len(self._filelist)
+        for i, file_ in enumerate(self._filelist):
+            # If the path is an s3 path set the respective element of self.is_s3 to True
+            if file_.startswith('s3'):
+                self.is_s3[i] = True
+            product_dict[file_] = is2ref.extract_product(file_, auth=self.auth)
```

Comment:
Haven't tested it yet, but are we going to run into the same auth issue (see [this fix](https://github.com/icesat2py/icepyx/pull/464/commits/454e30b2ecfe33c4ea0a3d240481db59fd0852ce)) we did with the variables module? An easy fix would be to indent this line and add an else with `auth=None` instead of `auth=self.auth`.

Reply:
Good catch! Just pushed a fix along these lines